homeassistant/components/sensor/uber.py

```diff
@@ -0,0 +1,247 @@
+                'price_details'
+        ).get(
+            'distance_unit'
+        ) is None or self._product.get(
+            'distance'
+        ) is None:
+            distance_key = 'Trip distance'
+            distance_val = 'N/A'
+        return {
+            'Product ID': self._product.get('product_id'),
+            'Product short description': self._product.get(
+                'short_description'),
+            'Product display name': self._product.get('display_name'),
+            'Product description': self._product.get('description'),
+            'Pickup time estimate (in seconds)':
+            self._product.get('time_estimate_seconds'),
+            'Trip duration (in seconds)': self._product.get('duration', 'N/A'),
+            distance_key: distance_val,
+            'Vehicle Capacity': self._product.get('capacity'),
+            'Minimum price': self._product.get('price_details').get('minimum'),
+            'Cost per minute': self._product.get(
+                'price_details').get('cost_per_minute'),
+            'Distance units': self._product.get(
+                'price_details').get('distance_unit'),
+            'Cancellation fee': self._product.get(
+                'price_details').get('cancellation_fee'),
+            'Cost per distance unit': self._product.get(
+                'price_details').get('cost_per_distance'),
+            'Base price': self._product.get('price_details').get('base'),
+            'Price estimate': self._product.get(
+                'price_details').get('estimate', 'N/A'),
+            'Price currency code': self._product.get(
+                'price_details').get('currency_code'),
+            'High price estimate': self._product.get(
+                'price_details').get('high_estimate', 'N/A'),
+            'Low price estimate': self._product.get(
+                'price_details').get('low_estimate', 'N/A'),
+            'Surge multiplier': self._product.get(
+                'price_details').get('surge_multiplier', 'N/A')
+        }
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return ICON
+
+    # pylint: disable=too-many-branches
+    def update(self):
+        """Get the latest data from the Uber API and update the states."""
+        self.data.update()
+        self._product = self.data.products[self._product_id]
+
+
+class LatLong(object):
+    """A container for a latitude and longitude pair."""
+
+    def __init__(self, latitude, longitude):
+        """Initialize the LatLong object."""
+        self.latitude = latitude
+        self.longitude = longitude
+
+
+# pylint: disable=too-few-public-methods
+class UberEstimate(object):
+    """The class for handling the time and price estimate."""
+
+    def __init__(self, session, start_latlong, end_latlong=None):
+        """Initialize the UberEstimate object."""
+        self._session = session
+        self.start_latitude = start_latlong.latitude
```

Comment:
@balloob Totally agree, the reason I did that is because I was getting a pylint error about having too many arguments (6 instead of 5 maximum) and was only able to squelch the warning by modifying the pylintrc as I couldn't get the `# pylint` comment syntax working for `max-args`.